api_open_fga.go

```diff
@@ -5684,58 +3900,61 @@ func (a *OpenFgaApiService) WriteAuthorizationModel(ctx context.Context, storeId
-		)
+	var (
+		requestStarted      = time.Now()
+		operationName       = "WriteAuthorizationModel"
+		localVarHTTPMethod  = http.MethodPost
+		localVarPostBody    interface{}
+		localVarReturnValue WriteAuthorizationModelResponse
+	)
 
-		localVarPath := "/stores/{store_id}/authorization-models"
-		if r.storeId == "" {
-			return localVarReturnValue, nil, reportError("storeId is required and must be specified")
-		}
+	localVarPath := "/stores/{store_id}/authorization-models"
+	if r.storeId == "" {
+		return localVarReturnValue, nil, reportError("storeId is required and must be specified")
+	}
 
-		localVarPath = strings.Replace(localVarPath, "{"+"store_id"+"}", url.PathEscape(parameterToString(r.storeId, "")), -1)
+	localVarPath = strings.Replace(localVarPath, "{"+"store_id"+"}", url.PathEscape(parameterToString(r.storeId, "")), -1)
 
-		localVarHeaderParams := make(map[string]string)
-		localVarQueryParams := url.Values{}
-		if r.body == nil {
-			return localVarReturnValue, nil, reportError("body is required and must be specified")
-		}
+	localVarHeaderParams := make(map[string]string)
+	localVarQueryParams := url.Values{}
+	if r.body == nil {
+		return localVarReturnValue, nil, reportError("body is required and must be specified")
+	}
 
-		// to determine the Content-Type header
-		localVarHTTPContentTypes := []string{"application/json"}
+	// to determine the Content-Type header
+	localVarHTTPContentTypes := []string{"application/json"}
 
-		// set Content-Type header
-		localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
-		if localVarHTTPContentType != "" {
-			localVarHeaderParams["Content-Type"] = localVarHTTPContentType
-		}
+	// set Content-Type header
+	localVarHTTPContentType := selectHeaderContentType(localVarHTTPContentTypes)
+	if localVarHTTPContentType != "" {
+		localVarHeaderParams["Content-Type"] = localVarHTTPContentType
+	}
 
-		// to determine the Accept header
-		localVarHTTPHeaderAccepts := []string{"application/json"}
+	// to determine the Accept header
+	localVarHTTPHeaderAccepts := []string{"application/json"}
 
-		// set Accept header
-		localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
-		if localVarHTTPHeaderAccept != "" {
-			localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
-		}
-		// body params
-		localVarPostBody = r.body
-		req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams)
-		if err != nil {
-			return localVarReturnValue, nil, err
-		}
+	// set Accept header
+	localVarHTTPHeaderAccept := selectHeaderAccept(localVarHTTPHeaderAccepts)
+	if localVarHTTPHeaderAccept != "" {
+		localVarHeaderParams["Accept"] = localVarHTTPHeaderAccept
+	}
+	// body params
+	localVarPostBody = r.body
+	req, err := a.client.prepareRequest(r.ctx, localVarPath, localVarHTTPMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams)
```

Comment:
Given that this has moved out of the retry loop, is there any potential concern around reusing the req? The [docs state](https://pkg.go.dev/net/http#RoundTripper) `Callers should not mutate or reuse the request until the Response's Body has been closed.`

Reply:
But don't you think in each loop the response body will have been closed before the next loop starts?

I can move it back to the loop, but I was trying to avoid all this extra compute

Reply:
Ah yeah sorry I didn't look close enough, we [call the API](https://github.com/openfga/go-sdk/blob/fix/retry-network-errors-token-issuer/api_open_fga.go#L3950) and then [read and close the body](https://github.com/openfga/go-sdk/blob/fix/retry-network-errors-token-issuer/api_open_fga.go#L3965-L3967) so I think that should mean we're safe to reuse